source/Calamari.Azure/Kubernetes/Discovery/AzureKubernetesDiscoverer.cs

```diff
@@ -30,10 +28,34 @@ public AzureKubernetesDiscoverer(ILog log) : base(log)
 
         public override IEnumerable<KubernetesCluster> DiscoverClusters(string contextJson)
         {
-            if (!TryGetDiscoveryContext<AccountAuthenticationDetails<AzureServicePrincipalAccount>>(contextJson, out var authenticationDetails, out _))
-                return Enumerable.Empty<KubernetesCluster>();
-            
-            var account = authenticationDetails.AccountDetails;
+            string accountType = string.Empty;
+            try
+            {
+                accountType = JToken.Parse(contextJson).SelectToken("authentication").SelectToken("authenticationMethod").ToString();
+            }
+            catch (Exception e)
+            {
+                Log.Warn("Could not read authentication method of target discovery context.");
```

Comment:
Looks like this is missing an exit 
```suggestion
                Log.Warn("Could not read authentication method of target discovery context.");
                return Enumberable.Empty<KubernetesCluster>();
```